backend/src/core/annotation_unit.py

```diff
@@ -30,20 +30,47 @@ def get_missing_dependencies(self):
 
         return missing_dependencies
 
+    def ready_for_annotation(self, annotation_value, missing_dependency):
+        """
+        Checks for annotation unit is ready for annotation
+        Gets the annotation value for dependcies from the genomic unit collection
+        """
+        ready = True
+        if annotation_value:
+            self.genomic_unit[missing_dependency] = annotation_value
+        else:
+            ready = False
+        return ready
+
+    def should_continue_annotation(self):
```

Comment:
Since the 'get_missing_dependencies' returns the python type "list," I recommend the missing_dependencies variable also return either an array; or, if you want the return type to be a string, you could do something like, `', '.join(missing_dependencies)` to create a comma delimited string with the missing dependencies.

Reply:
Will add this, and have it be set as a python type "list".

Reply:
Looks great! I see it now.